lib/em-jack/connection.rb

```diff
@@ -284,6 +284,11 @@ def disconnect
       @close_df = EM::DefaultDeferrable.new
     end
 
+    def force_reconnect
+      @intentionally_closed = false
```

Comment:
Is there a reason (other than an abundance of caution) that you set `@intentionally_closed` before forcing a reconnect? It can only be `true` if `disconnect` has been called in the past.

I see a few other suggestions in the code which suggest that no operations are valid on the `EMJack::Connection` after `disconnect` has been called. For example `reconnect!` doesn't clean the state of `intentionally_disconnected` either. Is it valid to call `disconnect` and then later `reconnect!`? If it is, then it's likely buggy and should be fixed as well.

But I suspect that it isn't, and so calling `force_reconnect` after `disconnect` should probably leave the value of `@intentionally_closed` alone.

Reply:
I think your initial assumption of abundant caution is accurate. I'll rectify.